src/databricks/labs/remorph/snow/databricks.py

```diff
@@ -299,6 +293,38 @@ def _parse_date_trunc(self: Databricks.Generator, expression: local_expression.D
     return self.func("TRUNC", expression.this, expression.args.get("unit"))
 
 
+def _get_within_group_params(
+    expr: exp.ArrayAgg | exp.GroupConcat,
+    within_group: exp.WithinGroup,
+) -> WithinGroupParams:
+    has_distinct = isinstance(expr.this, exp.Distinct)
+    agg_col = expr.this.expressions[0] if has_distinct else expr.this
+    order_expr = within_group.expression
+    order_col = order_expr.expressions[0].this
+    desc = order_expr.expressions[0].args.get("desc")
+    is_order_asc = not desc or exp.false() == desc
+    if has_distinct and agg_col != order_col:
+        raise ParseError("If both DISTINCT and WITHIN GROUP are specified, both must refer to the same column.")
```

Comment:
Can we leave a TODO here so that we can relax this condition in the future

Reply:
@nsenno-dbr This restriction is from Snowflake as mentioned in https://docs.snowflake.com/en/sql-reference/functions/array_agg#usage-notes

Reply:
Added a TODO comment